### DISCUSSION TITLE
Unable to generate coredump file When google test crashes

### DISCUSSION THREAD

**Discussion by xiaofyao** [google/googletest#3977] [Community Help]
Hi,
I need to use coredump file to debug when the program crashe, but the coredump file can't be generated.
Repetition steps:
1.  ulimit -c unlimited
2. execute the gtest executable file and crash

I've added the relevant compile support command to Cmake, but it still doesn't work:
```cmake
SET(CMAKE_BUILD_TYPE RelWithDebInfo)
SET(CMAKE_BUILD_TYPE "Debug")  
SET(CMAKE_CXX_FLAGS_DEBUG "$ENV{CXXFLAGS} -O0 -Wall -g2 -ggdb")  
SET(CMAKE_CXX_FLAGS_RELEASE "$ENV{CXXFLAGS} -O3 -Wall")  
add_compile_options(-std=c++11 -pthread -g  -ggdb -O0 -Wall   -fstack-protector-all -march=native -fnon-call-exceptions)
```

I wonder if GTest currently supports the generation of coredump files. If there is such a generation mechanism, please tell me how to use it.

By the way, why does Gtest need to run all the previous test cases before it can stop at the GDB breakpoint?

Br,
xiaofyao

**Comment by derekmauro** [comment]
There is nothing in GTest that prevents coredumps. I'm guessing it is just a configuration issue on your end.

You can use the `--gtest_filter` flag to only run specific tests: https://github.com/google/googletest/blob/main/docs/advanced.md#running-a-subset-of-the-tests
